pkg/testframework/framework.go

```diff
@@ -57,3 +74,61 @@ func DeleteCompletely(getObject func() (metav1.Object, error), deleteObject func
 		return obj.GetUID() != uid, nil
 	})
 }
+
+// IsResourceAvailable checks if tested resource is available(recreated by console-ioerator),
+// during 10 second period. If not error will be returned.
+func IsResourceAvailable(errChan chan error, client *Clientset, resource string) {
+	counter := 0
+	err := wait.Poll(1*time.Second, AsyncOperationTimeout, func() (stop bool, err error) {
+		err = getResource(client, resource)
+		if err == nil {
+			return true, nil
+		}
+		if counter == 10 {
+			if err != nil {
+				return true, fmt.Errorf("deleted console %s was not recreated", resource)
+			}
+			return true, nil
+		}
+		counter++
+		return false, nil
+	})
+	errChan <- err
+}
+
+// IsResourceUnavailable checks if tested resource is unavailable(not recreated by console-ioerator),
+// during 10 second period. If not error will be returned.
+func IsResourceUnavailable(errChan chan error, client *Clientset, resource string) {
+	counter := 0
+	err := wait.Poll(1*time.Second, AsyncOperationTimeout, func() (stop bool, err error) {
+		err = getResource(client, resource)
+		if err == nil {
+			return true, fmt.Errorf("deleted console %s was recreated", resource)
+		}
+		if errors.IsNotFound(err) {
+			counter++
+		}
+		if counter == 10 {
+			return true, nil
+		}
+		return false, nil
+	})
+	errChan <- err
+}
+
+func getResource(client *Clientset, resource string) error {
```

Comment:
Thinking about this:

```golang
// current unnamed
getResource(client, 'ConfigMap')
// vs named?
getResource(client, 'configmap', 'console-config')
// vs specific-to-client funcs?
getConfigmapByName('console-config')
getSecretByName('oauth-config')
```

Reply:
Makes sense to have resource based getters :+1:

Reply:
Realized it doesn't actually get the resource, it returns only an error.  So if you don't want to make specific ones, since this is more of a helper, it could be something like `resourceExists()`